### DISCUSSION TITLE
zkSync Plugin Error When Deploying Using Remix

### DISCUSSION THREAD

**Discussion by Play3rZer0** [Cyfrin/foundry-full-course-cu#2401] [General]
I am not sure if anyone else is getting this problem. I am able to successfully compile my source file in the contracts folder in Remix. I was able to deploy a contract to the zkSync Sepolia testnet blockchain before. Then when I tried to deploy a new contract, I get this error message:

Error: "underlying network changed (event=\"changed\", network={\"chainId\":324,\"name\":\"zkSync Era\"}, detectedNetwork={\"name\":\"sepolia\",\"chainId\":11155111,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)"

I have the wallet configured for zkSync Sepolia testnet and it is setup correctly on Metamask. How can I get Remix to detect the correct network?

**Comment by EngrPips** [comment]
Hello @Play3rZer0, I haven't run into this issue before, and I am unsure what the problem is, but deploying to `zksync` is optional, and you can skip it if it is causing you too much headache.

**Reply by Play3rZer0** [reply]
I found a way around it. I used the Brave browser with Metamask. It worked fine and I was able to deploy the contract to zkSync Sepolia test network using the plugin. However, I was able to do the same thing with Chrome/Metamask when I deployed my first contract to the zkSync test net. It only stopped detecting the zkSync network the next time I tried to deploy a new contract to the test network. Perhaps I need to reset Metamask transaction nonce, I am not so sure yet.

**Reply by EngrPips** [reply]
It's fascinating that you find a way around it. Keep crushing it.

**Reply by EmmanuelAjobo** [reply]
It is not working for me and i am using brave 

> I found a way around it. I used the Brave browser with Metamask. It worked fine and I was able to deploy the contract to zkSync Sepolia test network using the plugin. However, I was able to do the same thing with Chrome/Metamask when I deployed my first contract to the zkSync test net. It only stopped detecting the zkSync network the next time I tried to deploy a new contract to the test network. Perhaps I need to reset Metamask transaction nonce, I am not so sure yet.

It is not working for me and i am using brave

**Reply by EngrPips** [reply]
Maybe try another browser.

**Comment by wangshouren7** [comment]
Hello, I just meet this problem. For me, after i clear browser cache(localstorage, cookies, indexdb...), it works. Remember backup your files, because this operation will clear all the changes you made in Remix.

**Reply by EngrPips** [reply]
Interesting, Thanks for the heads-up, buddy.

**Comment by 0xSaksham** [comment]
@Play3rZer0 Try using remote devnet, it works on zksync plugin!

**Comment by 0xSaksham** [comment]
Hey @Play3rZer0 you have to change the wallet network to zksync sepolia in the lower right selection box

**Comment by artitaya-git** [comment]
Hello, I'm facing the same issue. Closely following the tutorial but cannot connect to a local devnet. Changed browser and cleared cache but didn't work. Please help
![Screenshot 2025-01-11 193041](https://github.com/user-attachments/assets/986b7a46-a9bd-4ddf-a995-1d6a87c1f762)

**Reply by EngrPips** [reply]
Click on the `Local Devnet` and it should automatically select an account after the dropdown opens.

**Comment by KMean** [comment]
> Hello, I just meet this problem. For me, after i clear browser cache(localstorage, cookies, indexdb...), it works. Remember backup your files, because this operation will clear all the changes you made in Remix.

Also pinning your workspace/contract in remix should remember your files.
